File: app/src/test/java/medlife/com/br/adapter/CategoryAdapterTest.java
package medlife.com.br.adapter;

import static org.junit.Assert.*;
import static org.mockito.Mockito.*;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;
import androidx.recyclerview.widget.RecyclerView;
import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import medlife.com.br.R;
import medlife.com.br.model.Category;
import java.util.ArrayList;
import java.util.List;

public class CategoryAdapterTest {
    @Mock
    Context mockContext;
    @Mock
    View mockView;
    @Mock
    ViewGroup mockParent;
    @Mock
    ImageView mockImageView;
    @Mock
    TextView mockTextView;

    private CategoryAdapter categoryAdapter;
    private List<Category> categoryList;
    private Category category;

    @Before
    public void setUp() {
        MockitoAnnotations.initMocks(this);
        
        // Setup test data
        category = new Category(123, "Medicamentos");
        category.setName("Medicamentos");
        category.setImageResId(123);
        
        categoryList = new ArrayList<>();
        categoryList.add(category);
        
        categoryAdapter = new CategoryAdapter(mockContext, categoryList);
    }

    @Test
    public void testConstructor() {
        assertNotNull(categoryAdapter);
        assertEquals(1, categoryAdapter.getItemCount());
    }

    @Test
    public void testGetItemCount() {
        assertEquals(1, categoryAdapter.getItemCount());
    }

    @Test
    public void testGetItemCount_EmptyList() {
        CategoryAdapter emptyAdapter = new CategoryAdapter(mockContext, new ArrayList<>());
        assertEquals(0, emptyAdapter.getItemCount());
    }

    @Test
    public void testCategoryViewHolderConstructor() {
        when(mockView.findViewById(R.id.imageSubcategory)).thenReturn(mockImageView);
        when(mockView.findViewById(R.id.textSubcategory)).thenReturn(mockTextView);

        CategoryAdapter.CategoryViewHolder holder = new CategoryAdapter.CategoryViewHolder(mockView);
        
        assertNotNull(holder.imageSubcategory);
        assertNotNull(holder.textSubcategory);
    }
} 

File: app/src/test/java/medlife/com/br/adapter/CategoryAdapterTest.kt
package medlife.com.br.adapter

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import medlife.com.br.R
import medlife.com.br.model.Category
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Before
import org.junit.Test
import org.mockito.Mock
import org.mockito.Mockito.`when`
import org.mockito.MockitoAnnotations

class CategoryAdapterTest {
    @Mock lateinit var mockContext: Context
    @Mock lateinit var mockView: View
    @Mock lateinit var mockParent: ViewGroup
    @Mock lateinit var mockImageView: ImageView
    @Mock lateinit var mockTextView: TextView

    private lateinit var categoryAdapter: CategoryAdapter
    private lateinit var categoryList: MutableList<Category>
    private lateinit var category: Category

    @Before
    fun setUp() {
        MockitoAnnotations.initMocks(this)

        category = Category(123, "Medicamentos")
        category.name = "Medicamentos"
        category.imageResId = 123

        categoryList = mutableListOf(category)
        categoryAdapter = CategoryAdapter(mockContext, categoryList)
    }

    @Test
    fun testConstructor() {
        assertNotNull(categoryAdapter)
        assertEquals(1, categoryAdapter.itemCount)
    }

    @Test
    fun testGetItemCount() {
        assertEquals(1, categoryAdapter.itemCount)
    }

    @Test
    fun testGetItemCount_EmptyList() {
        val emptyAdapter = CategoryAdapter(mockContext, mutableListOf())
        assertEquals(0, emptyAdapter.itemCount)
    }

    @Test
    fun testCategoryViewHolderConstructor() {
        `when`(mockView.findViewById<ImageView>(R.id.imageSubcategory)).thenReturn(mockImageView)
        `when`(mockView.findViewById<TextView>(R.id.textSubcategory)).thenReturn(mockTextView)

        val holder = CategoryAdapter.CategoryViewHolder(mockView)

        assertNotNull(holder.imageSubcategory)
        assertNotNull(holder.textSubcategory)
    }
}
